framework/src/BaseComponent.ts

```diff
@@ -1,18 +1,17 @@
-import { UnknownObject } from '@jovotech/common';
+import { DeepPartial, UnknownObject } from '@jovotech/common';
 import { ComponentData, JovoComponentInfo } from './index';
 import { Jovo } from './Jovo';
 import { JovoProxy } from './JovoProxy';
 import { ComponentOptions } from './metadata/ComponentMetadata';
 
+export type ComponentConfig<COMPONENT extends BaseComponent = BaseComponent> =
+  COMPONENT['$component']['config'];
+
 export type ComponentConstructor<COMPONENT extends BaseComponent = BaseComponent> = new (
   jovo: Jovo,
-  options?: ComponentOptions<COMPONENT>,
```

Comment:
I'm not sure if I'm mistaken, but shouldn't we allow to pass all `options` to the constructor, not just the `config`?

Reply:
We only need the configuration when initializing the component itself, the other metadata is not used by the component instance itself.

Reply:
Thanks. I'm still not sure. Isn't `name` an important part of the `options` as well?

```
components: [MenuComponent, new ComponentDeclaration(MenuComponent2, { name: 'MenuComponent2' })],
```

Or is this still possible?

Reply:
Good catch, in some edge cases it could be actually helpful to have access to the other options. I will change it accordingly.